rfc/rfc-66/rfc-66.md

```diff
@@ -0,0 +1,246 @@
+but the write_token has deterministic priority in sorting, which breaks the file generation order.
+
+![log file sequence](log_file_sequence.png)
+
+### Format changes
+
+| Type                       | Changes                                                                                                                            |
+|----------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| Commit/compaction metadata | No changes                                                                                                                         |
+| Commit file name           | we are only going to add completion time in completed metadata filename for all actions                                            |
+| Log file format            | Adding deltacommit instant time to log file name. The file name does not contain base instant time; No changes to log block format |
+
+### Commit Protocol
+
+| Type                            | Content                                                                                                                                       |
+|---------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| Writer expectations             | Writer does not need to look up the latest file slice to fetch the base commit time anymore, it always uses the current instant time directly |
+| Conflict Resolution             | No explicit conflict resolution, it is handled by the compactor                                                                               |
+| Reader expectations             | Readers still query based on file slices, see the pseudocode for new file slicing                                                             |
+| Synchronization and correctness | Still need some mutex on commit log (timeline). Getting the completion time and adding to the timeline needs to happen atomically             |
+
+**Note:** Clustering and updates conflict resolution is not in the scope of non-blocking concurrency control, you still need the OCC for course-grained failures and retries. 
+
+## Appendix
+
+### Pseudocode for New File Slicing
+
+```python
+# new file slicing
+file_slice_barriers = []
+# sort the base file by start time reversely and get the barriers
+for base_file in fg.getBaseFilesSortedReverselyByStartTime():
+  file_slice_barriers.add(instant_time(base_file))
+
+file_slices = []
+for log_file in fg.getLogFilesSortedByCompletionTime():
+  completion_time = completion_time(log_file)
+  file_slice = find_file_slice(completion_time, file_slice_barriers)
+  file_slices.add(file_slice)
+
+def find_file_slice(completion_time, file_slice_barriers):
+  # Note: needs to handle special case where file_slice_barriers is empty (no base file)
```

Comment:
Let's complete this part. Barrier and filegroup id need to have one-one mapping. If there are no base files, the instant time of the earliest log file can be the barrier, and so all log files get added to the same slice.

Reply:
You are right, thanks for the clarification ~